client/state/upgrade_int_test.go

```diff
@@ -109,9 +122,59 @@ func TestBoltStateDB_UpgradeOld_Ok(t *testing.T) {
 			}
 			require.NoError(t, db.PutDevicePluginState(ps))
 
+			registry, err := db.GetDynamicPluginRegistryState()
+			require.Nil(t, registry)
+
+			require.NoError(t, err)
 			require.NoError(t, db.Close())
 		})
 	}
+
+	t.Run("testdata/state-1.2.6.db.gz", func(t *testing.T) {
```

Comment:
Note to reviewers: this file got created from the following test code on a 1.2.6 branch:

<details><summary>TestBoltStateDB_WriteTestData</summary>

```go

func TestBoltStateDB_WriteTestData(t *testing.T) {
	os.Mkdir("newdata", 0775)
	dbI, err := NewBoltStateDB(testlog.HCLogger(t), "newdata")
	require.NoError(t, err)

	db := dbI.(*BoltStateDB)
	defer db.Close()

	ps := &dynamicplugins.RegistryState{
		Plugins: map[string]map[string]*dynamicplugins.PluginInfo{
			"csi-controller": {
				"plugin-1": {
					Name:    "plugin-1",
					Type:    string(structs.CSIPluginTypeMonolith),
					Version: "v0.1",
					ConnectionInfo: &dynamicplugins.PluginConnectionInfo{
						SocketPath: "/var/nomad/client/csi/plugin-1/csi.sock",
					},
					AllocID: "489cd5cf-3cfc-6792-3465-822371765b32",
					Options: map[string]string{
						"Provider":            "demo",
						"MountPoint":          "/var/nomad/client/csi/plugin-1",
						"ContainerMountPoint": "/csi",
					},
				},
			},
			"csi-node": {
				"plugin-1": {
					Name:    "plugin-1",
					Type:    string(structs.CSIPluginTypeMonolith),
					Version: "v0.1",
					ConnectionInfo: &dynamicplugins.PluginConnectionInfo{
						SocketPath: "/var/nomad/client/csi/plugin-1/csi.sock",
					},
					AllocID: "489cd5cf-3cfc-6792-3465-822371765b32",
					Options: map[string]string{
						"Provider":            "demo",
						"mountpoint":          "/var/nomad/client/csi/plugin-1",
						"ContainerMountPoint": "/csi",
					},
				},
				"plugin-2": {
					Name:    "plugin-2",
					Type:    string(structs.CSIPluginTypeNode),
					Version: "v0.2",
					ConnectionInfo: &dynamicplugins.PluginConnectionInfo{
						SocketPath: "/var/nomad/client/csi/plugin-2/csi.sock",
					},
					AllocID: "68221094-25a6-eeb6-8490-c71a03d46b0f",
					Options: map[string]string{
						"Provider":            "other-demo",
						"MountPoint":          "/var/nomad/client/csi/plugin-2",
						"ContainerMountPoint": "/csi",
					},
				},
			},
		},
	}
	db.PutDynamicPluginRegistryState(ps)

}
```

</details>